frontend/src/ClassDetail.svelte

```diff
@@ -82,6 +85,26 @@ Total points: ${assignmentPoints.toFixed(2)}/${totalMaximumPoints}
 Average points: ${average}`;
 }
 
+function createTaskTypeSummary(clazz) {
```

Comment:
This would be quite error-prone (as can be seen by the wrong per-student point totals). I would rather have a single function that returns the currently active tasks (after filtering), and then render the table and compute all points and statistics based on the active task list.